Includes/findXBE.cpp

```diff
@@ -21,15 +22,35 @@ int findXBE(void* list) {
       sprintf(tmp, "%s%s\\default.xbe", path, fData.f_FileName);
       tmpFILE = fopen(tmp, "rb");
       if (tmpFILE != nullptr) {
-        xbeMenuItem* itm = new xbeMenuItem(fData.f_FileName, tmp);
-        gmi_list->push_back(*itm);
+        read_bytes = fread(xbeData, 1, SECTOR_SIZE, tmpFILE);
+        XBE *xbe = (XBE*)xbeData;
+        if(xbe->type != XBETYPESTRING ||
+           xbe->baseAddress != BASEADDRESS ||
+           xbe->baseAddress > xbe->certAddress ||
+           xbe->certAddress + 4 >= (xbe->baseAddress + xbe->sizeOfHeaders) ||
+           xbe->sizeOfHeaders > read_bytes) {
+          continue;
+        }
+        XBECert *xbeCert = (XBECert*)&xbeData[xbe->certAddress -
+                                              xbe->baseAddress];
+        int offset = 0;
+        while(offset < XBENAMESIZE) {
+          sprintf(&xbeName[offset], "%c", (char)xbeCert->xbeName[offset]);
```

Comment:
This will fail for `xbeCert->xbeName[offset] >= 0x100`; as long as this is mostly used in western countries this is fine, but you'll need an issue for this and it should be documented. You could append a "?" symbol for unsupported characters.